core/amber/src/main/scala/edu/uci/ics/amber/engine/common/client/AmberClient.scala

```diff
@@ -68,18 +73,27 @@ class AmberClient(system: ActorSystem, workflow: Workflow, controllerConfig: Con
       "get observable with a remote client actor is not supported"
     )
     val clazz = ct.runtimeClass
-    if (registeredObservables.contains(clazz)) {
-      return registeredObservables(clazz).asInstanceOf[Observable[T]]
-    }
-    val sub = Subject[T]
-    val req = ObservableRequest({
-      case x: T =>
-        sub.onNext(x)
+    val observable =
+      if (registeredObservables.contains(clazz)) {
+        registeredObservables(clazz).asInstanceOf[Observable[T]]
+      } else {
+        val sub = Subject[T]
+        val req = ObservableRequest({
+          case x: T =>
+            sub.onNext(x)
+        })
+        Await.result(clientActor ? req, 2.seconds)
```

Comment:
what does this 2 seconds do? can you make it a variable to improve readability? also make it configurable?

Reply:
also would this `Await.result` break?

Reply:
the `2 seconds` is the timeout for this request. What do you mean by break? This is a blocking call and it will throw an error upon timeout or anything bad happens.